4-loops/Review.swift

```diff
@@ -11,20 +11,31 @@ for num in 1...100 {
 }
 
 
-// Second Challenge 
-// checkPrime can be changed to a positive number greater than 1
-var checkPrime = 17
-// Assume the number is prime until proven otherwise
-var isPrime = true
-for num in 2...checkPrime - 1 {  
-  // If checkPrime is fully divisible by the current number it's not a prime number
-  if checkPrime % num == 0 {
-    isPrime = false
-    break
+// Second Challenge
+// checkPrime can be any integer
+var checkPrime = 5 
+// isPrime may be initialized with either a true or false value
+var isPrime = false
```

Comment:
Hey @jdwagner17, curious why this was changed, from true to false. 

If we keep it as true, then we could avoid adding the else statement and block in line 25.

Reply:
I think it may be "best practice" to do it the way you suggested. 

I included this because it may help new programmers better understand the relationship. If the variable is something we don't control (for whatever reason), they will know how to compensate for an unknown. That may not exist in Swift, as I noticed I could not do

var checkPrime = Bool

The situation I am imagining may not exist.

Reply:
var isPrime: Bool makes sense to me, but I get an error when used in CodeAcademy.

Review.swift:64:43: error: variable 'isPrime' used before being initialized
print("Is \(checkPrime) a prime number? \(isPrime)!")  
                                          ^
Review.swift:44:5: note: variable defined here
var isPrime: Bool
    ^

I think I understand your comment about iterations within the loop. Will fix in commit

Reply:
Yep, I think this is something happening under the hood, where Swift's interpreter sees a situation where `isPrime` might not get assigned at all before the print statement. 

Try fixing the assignment issue of `isPrime` being assigned during each iteration and that might resolve this issue as well :)

Reply:
And these things happen! 

I'd still say trying to debug on your own (with print statements) is helpful and useful later down the line too.
And also, be kind to yourself!